src/js/me-i18n.js

```diff
@@ -1,169 +1,223 @@
-            if (replace.hasOwnProperty(character)) {
-                regex = new RegExp(character, 'g');
-                str = str.replace(regex, replace[character]);
-            }
-        }
-        return str;
-    };
-
-    /**
-     * Translate strings to the page language or a given language.
-     *
-     *
-     * @param uid
-     *   A string containing a unique id of the translated text to retrieve.
-     *
-     * @param options
-     *   - 'context' (defaults to the default context): The context the source string
-     *     belongs to.
-     *
-     * @return
-     *   The translated string, escaped via i18n.methods.checkPlain()
-     */
-    i18n.methods.t = function (uid, options) {
-        var str;
-
-        // Fetch the localized version of the string.
-        if (i18n.locale.strings && i18n.locale.strings[options.context]) {
-            str = i18n.locale.strings[options.context][uid];
-        }
-
-        // Fallback to default language if requested uid is not translated
-        if (!str && i18n.locale.strings && i18n.locale.strings[i18n["default"]]) {
-            str = i18n.locale.strings[i18n["default"]][uid];
-        }
-
-        // As a last resort, use the requested uid, to mimic original behavior of i18n utils (in which uid was the english text)
-        str = str || uid;
-
-        return i18n.methods.checkPlain(str);
-    };
-
-
-    /**
-     * Wrapper for i18n.methods.t()
-     *
-     * @see i18n.methods.t()
-     * @throws InvalidArgumentException
-     */
-    i18n.t = function(str, options) {
-
-        if (typeof str === 'string' && str.length > 0) {
-
-            // check every time due language can change for
-            // different reasons (translation, lang switcher ..)
-            var language = i18n.getLanguage();
-
-            options = options || {
-                "context" : language
-            };
-
-            return i18n.methods.t(str, options);
-        }
-        else {
-            throw {
-                "name" : 'InvalidArgumentException',
-                "message" : 'First argument is either not a string or empty.'
-            };
-        }
-    };
-
-// end i18n
-    exports.i18n = i18n;
-}(document, mejs));
-
-// i18n fixes for compatibility with WordPress
-;(function(exports, undefined) {
-
-    "use strict";
-
-    if ( typeof mejsL10n != 'undefined' ) {
-        exports[mejsL10n.language] = mejsL10n.strings;
-    }
-
-}(mejs.i18n.locale.strings));
+(function (doc, win, mejs, undefined) {
+
+	var i18n = {
+		/**
+		 * @type {String}
+		 */
+		default: 'en',
+
+		/**
+		 * @type {String[]}
+		 */
+		locale: {
+			language: (mejs.i18n && mejs.i18n.locale.language) || '',
+			strings: (mejs.i18n && mejs.i18n.locale.strings) || {}
+		},
+
+		/**
+		 * Filters for available languages
+		 *
+		 * @see https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_and_Plurals#List_of_Plural_Rules
+		 * @type {Object[]}
+		 */
+		rules: [
```

Comment:
This could be a simple string like
```js
exports.en = {
    'mejs.plural-form': 'nplurals=2; plural=(n > 1);',
   // ...
}
```

or 

```js
exports.en = {
    'mejs.plural-form': '0', // value can be 0-16 if you use the Mozilla version or 1-6 (0-5) if you use http://unicode.org/cldr/trac/browser/trunk/common/supplemental/plurals.xml
   // ...
}
```